### PR TITLE
folder structure, server, and start of database-need associations

### PR DESCRIPTION
**Pull request by mswil**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,32 @@
...
+## Contributing
+👪 Chris Winters/ Molly Wilson/ Joshua Wood/ Karsten Darden
+## Questions
+:octocat: Reach us on GitHub <br />
+<br />
+
+[cwinters87](https://github.com/cwinters87)<br />
+[mswil](https://github.com/mswil)<br />
+[Joshuawd190](https://github.com/Joshuawd190)<br />
+karsten darden <br />
```
Comment (mswil) [inline README.md:29]: Karsten's GitHub is kdarden45
```suggestion
[kdarden](https://github.com/kdarden45) <br />
```

File: models/Category.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+const sequelize = require('../config/connection.js');
+
+class Category extends Model {}
+
+Category.init(
+  {
+    id: {
+      type: DataTypes.INTEGER,
+      allowNull: false,
```
Comment (mswil) [inline models/Category.js:11]: Doesn't need to be changed, but allowNull is assumed to be false if the column has primaryKey: true.

File: models/Item.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+      primaryKey: true,
+      autoIncrement: true,
+      allowNull: false
+    },
+    name: {
+      type: DataTypes.STRING,
+      allowNull: false
+    },
+    description: {
+      type: DataTypes.STRING,
```
Comment (mswil) [inline models/Item.js]: .TEXT, it allows for more characters. .STRING is only 225 characters. Schema mockup has been updated to reflect this.
```suggestion
      type: DataTypes.TEXT,
```

File: models/Item.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+class Item extends Model {}
+
+// set up fields and rules for Item model
+Item.init(
+  {
+    id: {
+      type: DataTypes.INTEGER,
+      primaryKey: true,
+      autoIncrement: true,
+      allowNull: false
```
Comment (mswil) [inline models/Item.js:16]: See previous allowNull/primaryKey comment

File: models/Item.js [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+      allowNull: true
+    },
+    price: {
+      type: DataTypes.DECIMAL,
+      allowNull: false
+    },
+    stock: {
+      type: DataTypes.INTEGER,
+      allowNull: false,
+      defaultValue: 10,
```
Comment (mswil) [inline models/Item.js]: defaultValue should be 0
```suggestion
      defaultValue: 0,
```

File: models/Order.js [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+
+class Order extends Model {}
+
+Order.init(
+  {
+    id: {
+      type: DataTypes.INTEGER,
+      primaryKey: true,
+      autoIncrement: true,
+      allowNull: false
```
Comment (mswil) [inline models/Order.js:13]: See previous allowNull/primaryKey comment

File: models/Order.js [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+    },
+    user_id: {
+      type: DataTypes.STRING,
```
Comment (mswil) [inline models/Order.js:16]: DataType should be INTEGER
```suggestion
      type: DataTypes.INTEGER,
```

File: models/OrderItems.js [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+
+class OrderItems extends Model {}
+
+OrderItems.init(
+  {
+    id: {
+      type: DataTypes.INTEGER,
+      primaryKey: true,
+      autoIncrement: true,
+      allowNull: false
```
Comment (mswil) [inline models/OrderItems.js:13]: See previous allowNull/primaryKey comment

File: models/OrderItems.js [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+      type: DataTypes.INTEGER,
+      references: {
+        model: 'order',
+        key: 'id'
+      }      
+    },
+    amount_ordered: {
+      type: DataTypes.INTEGER,
+      allowNull: false,
+      defaultValue: 10,
```
Comment (mswil) [inline models/OrderItems.js]: defaultValue should be 1
```suggestion
      defaultValue: 1,
```

File: models/User.js [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+    return bcrypt.compareSync(loginPw, this.password)
+  }
+}
+
+// create fields/columns for User model
+User.init(
+  {
+    id: {
+      type: DataTypes.INTEGER,
+      allowNull: false,
```
Comment (mswil) [inline models/User.js:18]: See previous allowNull/primaryKey comment

File: models/User.js [hunk 11]
```diff
@@ -0,0 +1,60 @@
...
+      primaryKey: true,
+      autoIncrement: true
+    },
+    email: {
+      type: DataTypes.STRING,
+      allowNull: false
+    },
```
Comment (mswil) [inline models/User.js]: Should probably validate it is an email address

```suggestion
    email: {
      type: DataTypes.STRING,
      allowNull: false,
      validate: {
        isEmail: true
      }
    },
```

File: models/index.js [hunk 12]
```diff
@@ -0,0 +1,30 @@
...
+// Categories have many Items
+Category.hasMany(Item)
+// Products belongToMany Orders (through OrderItems)
+Item.belongsToMany(Order, {
+  through: OrderItems
+})
+// Order belongToMany Items (through OrderItems)
+Order.belongsToMany(Item, {
+  through: OrderItems
+})
```
Comment (mswil) [inline models/index.js]: UPDATED SUGGESTION
Looked at the documentation. I think this is all we need. 

```suggestion
// User
User.hasMany(Order);
Order.belongsTo(User);

Item.belongsToMany(Order, {
  through: OrderItems,
})
Order.belongsToMany(Item, {
  through: OrderItems,
})

// Order
Order.hasMany(OrderItems);
OrderItems.belongsTo(Order);

// OrderItems
OrderItems.belongsTo(Item);
Item.hasMany(OrderItems);

// Item
Item.belongsTo(Category);
Category.hasMany(Item);
```

**Review by cwinters87 — APPROVED** [review]

**Pull request merged**